### DISCUSSION TITLE
How to Build and testing one DLL

### DISCUSSION THREAD

**Discussion by ManuelLopezNavarrete** [quartznet/quartznet#2121] [Q&A]
Hi,

My situation:
Add modification in Quartz.Jobs.
Want to test the changes in my other project who is using Quartz, with all reference of Quartz pointing to the 3.7.0 on the official nuget. 
If i copy and paste the  Quartz.Jobs.dll manual of course don't work because the rest of quartz dll are diferent version.

I know can remove all nuget reference and add all quartz dlls manually, there are another best approach i'm missing? 

Thanks

**Comment by lahma** [comment]
I think it's easiest to develop your job/plugin outside of Quartz DLLs. Just define it inside your project which references Quartz and Quartz.Jobs (if needed).

**Comment by ManuelLopezNavarrete** [comment]
The ultimate problem is, I re use a lot of the logic of the actual DirectoryScanJob, only change the way of the get the list of files, and the object who represent each file FileInfo for SftpFile.
So i need not only the IJob, some internal and private classes to.

I think i will create a new console project on the Quartz sln (don't committed) add all references and doing from there.

**Reply by A9G-Data-Droid** [reply]
You can maintain your own copies of code in your plugin without maintaining the entire Quartz solution to do so.
